exploration/check_return_period_funcs.py

```diff
@@ -0,0 +1,179 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     custom_cell_magics: kql
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: ds-raster-stats
+#     language: python
+#     name: ds-raster-stats
+# ---
+
+# %%
+# %matplotlib inline
+# %load_ext autoreload
+# %autoreload 2
+
+# %%
+from src.utils import cloud_utils
+from src.utils import return_periods as rp
+import pandas as pd
+import numpy as np
+from io import BytesIO
+import pyarrow
+from azure.storage.blob import BlobServiceClient
+import re
+from scipy.stats import beta
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FuncFormatter
+
+def to_snake_case(name):
+    s1 = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", name)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+    
+sample_admin = "Awdal" # just used for plotting checking
+# %% [markdown]
+# Load data
+
+# %%
+pc = cloud_utils.get_container_client(mode="dev", container_name="projects")
+blob_name = "ds-floodscan-ingest/df_aer_sfed_som_adm1_zstats.parquet"
+blob_client = pc.get_blob_client(blob_name)
+blob_data = blob_client.download_blob().readall()
+df = pd.read_parquet(BytesIO(blob_data))
+df.columns = [to_snake_case(col) for col in df.columns]
+
+# %% [markdown]
+# ## Pre Process Data
+#
+# So just take yearly max values
+
+# %%
+df["year"] = pd.to_datetime(df["date"]).dt.year
+df_max = df.groupby(["adm1_en", "adm1_pcode", "year"]).max().reset_index()
+
+# %% [markdown]
+# ## Calculate LP3 Params
+#
+# Show the implementation by first just doing each method one by one on a single sample admin
+
+# %%
+rp.lp3_params(df_max[df_max["adm1_en"]==sample_admin].value,est_method = "usgs")
+
+
+# %%
+rp.lp3_params(df_max[df_max["adm1_en"]==sample_admin].value,est_method = "lmoments")
+
+# %%
+rp.lp3_params(df_max[df_max["adm1_en"]==sample_admin].value,est_method = "scipy")
+
+# %% [markdown]
+# We can also run them all at once with this wrapper
+
+# %%
+rp.lp3_params_all2(df_max[df_max["adm1_en"]==sample_admin].value)
```

Comment:
Getting an error since the `lp3_params_all2` function doesn't exist. Is this just `lp3_params_all`?

Reply:
yes sorry - leftover dev code - fixed [846210a](https://github.com/OCHA-DAP/ds-raster-stats/pull/7/commits/846210aff992e670bcc63151b59778180313387f)